### DISCUSSION TITLE
Showcase OS setup

### DISCUSSION THREAD

**Discussion by marcinjahn** [YaLTeR/niri#587] [Ideas]
Currently, to set up Niri, you need to do quite a bit of configuration to get to the usable state, and that's under the condition that you set up things properly and not encounter any issues.
I think it'd be quite valuable to have some form of a showcase setup that'd be easy to spin up, e.g. in a VM. I'm thinking something like a [Blue Build](https://blue-build.org/) setup maybe, where it'd be a declarative set of configs easy to get an ISO of and install. The method doesn't really matter that much, but the general ideal would be to allow people to easily check out Niri in some "ideal" setup, without the risk of bricking their systems, or giving them chance to misconfigure something and getting away with false bad impression.

The idea is inspired by me wanting to try Niri out, since the premise sounds awesome. Going through Getting Started is pretty discouraging, with all the things that need to be set up to get to some working environment. At the point where I'm reading Getting Started I'm not even sure if I'll like the end result, so it's a tough sell to convince me to go through the learning and setup process and spend time doing that.

**Comment by YaLTeR** [comment]
Yeah, I understand the problem and I agree; though, this is a problem with pretty much any WM that does not come with a desktop environment. In Fedora there are spins for certain WMs, like there's a [Fedora Sway Spin](https://fedoraproject.org/spins/sway/) with basic things pre-configured. Something in this format could work.

I'm not sure I want to do/maintain this myself though. But if there's interest, I can prioritize fixing some things that turn out to be necessary. E.g. I imagine config includes, at least in some basic form, are needed.

**Comment by sodiboo** [comment]
to be honest i think NixOS would be a good choice for this because (a) most of the "setup work" related to niri itself, e.g. xwayland-satellite, should just be part of my niri-flake, so there is very little duplicate work, and the overall "sample config" would be quite concise and could dedicate significant parts towards configuring niri and the desktop environment (b) nixos can easily generate a VM with any arbitrary config you throw at it, and (c) it could run as part of my CI and automatically get updated to the latest niri git versions

since most of the stuff you'd want in such a config is easily available in various consumers of niri-flake, it shouldn't really be too much work to get a decent config working, and once it is configured, there's little extra maintenance burden for me (as i said, most of keeping it up to date can just be automated).

really, something that could totally be done, is to just like, package various nixos configs with niri in them, as qemu VMs you can try. main issue would be to get niri working in a VM because i still haven't fucking figured out how to get real GPU emulation in the VMs produced by NixOS :sob:. but once that's done it would be pretty cool to be able to just hop around various users' configs. try 'em out. then there's basically no maintenance burden of keeping a Sample Intro Config because there isn't a main one, just "here's sodiboo's config. here's algernon's config. here's oli's config. here's bunbun's config. here's sugar's config. wanna try any of them? download button next to all of them! :D". this might be more interesting because it would allow you to see how real people use niri in their actual day-to-day lives

like, if this is something that sounds interesting to people, i think i'd be willing to maintain a NixOS version of it.

**Comment by sodiboo** [comment]
> i still haven't fucking figured out how to get real GPU emulation in the VMs produced by NixOS 😭

update: you can now run my config as a sample OS! just type `nix run github:sodiboo/system#nitrogen`. niri works in there!

**one config down**, however many else to go...
